pandas/core/dtypes/cast.py

```diff
@@ -246,6 +246,85 @@ def trans(x):
     return result
 
 
+def maybe_cast_result(result, obj, numeric_only: bool = False, how: str = ""):
+    """
+    Try to cast the result to the desired type,
+    we may have roundtripped through object in the mean-time.
+
+    If numeric_only is True, then only try to cast numerics
+    and not datetimelikes.
```

Comment:
can you add a Parameters / Returns here, and result: T and returns T (import from pandas._typing)